server.js

```diff
@@ -3905,6 +3906,268 @@ cache(function(data, match, sendBadge, request) {
+  const badgeData = getBadgeData('contributors', data);
+
+  gitlabHelpers.request(request, apiURL, {}, true, function(err, res, buf) {
+    if ((err != null) || (res.statusCode !== 200)) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+    } else {
+      try {
+        badgeData.text[1] = metric(parseInt(res.headers['x-total']));
+        badgeData.colorscheme = 'blue';
+        sendBadge(format, badgeData);
+      } catch(e) {
+        badgeData.text[1] = 'invalid';
+        sendBadge(format, badgeData);
+      }
+    }
+  });
+}));
+
+// GitLab build status integration
+camp.route(/^\/gitlab\/((https?)\/([^/]+)\/)?build\/([^/]+)\/([^/]+)(?:\/([^/]+))?\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  const user = match[4];
+  const repo = match[5];
+  const branch = match[6] || 'master';
+  const format = match[7];
+  const apiURL = gitlabHelpers.baseURL(match) + '/' + user + '/' + repo + '/badges/' + branch + '/pipeline.svg';
+  const badgeData = getBadgeData('build', data);
+
+  request(apiURL, function(err, res, buffer) {
+    if ((err != null) || (res.statusCode !== 200)) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+    } else {
+      xml2js.parseString(buffer.toString(), function (err, data) {
+        if (err != null) {
+          badgeData.text[1] = 'invalid';
+          sendBadge(format, badgeData);
+          return;
+        }
+        try {
+          const build_status = data.svg.g[1].g[0].text[3]['_'].trim();
+          badgeData.text[1] = build_status;
+          badgeData.colorscheme = gitlabHelpers.buildColor(build_status);
+          sendBadge(format, badgeData);
+        } catch(e) {
+          badgeData.text[1] = 'invalid';
+          sendBadge(format, badgeData);
+        }
+      });
+    }
+  });
+}));
+
+// GitLab issues & merge requests integration
+camp.route(/^\/gitlab\/((https?)\/([^/]+)\/)?(issue|merge-request)s(-closed)?(-raw)?\/([^/]+)\/(.+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  const isIssue = (match[4] === 'issue');
+  const isClosed = !!match[5];
+  const isRaw = !!match[6];
+  const inWords = (isClosed ? 'closed' : 'open');
+  const user = match[7];
+  const repo = match[8];
+  const format = match[9];
+  const apiURL = gitlabHelpers.baseURL(match) + '/api/v4/projects/' + user + '%2F' + repo + '/' + (isIssue ? 'issues' : 'merge_requests');
+  const badgeData = getBadgeData((isRaw ? (inWords + ' ') : '') + (isIssue ? 'issues' : 'merge requests'), data);
+
+  gitlabHelpers.request(request, apiURL, { state: (isClosed ? 'closed' : 'opened') }, true, function(err, res, buf) {
+    if ((err != null) || (res.statusCode !== 200)) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+    } else {
+      try {
+        badgeData.text[1] = metric(parseInt(res.headers['x-total'])) + (isRaw ? '' : (' ' + inWords));
```

Comment:
Shall we extract `parseInt(res.headers['x-total'])` to a local variable? It will make this line and the one below slightly more readable.